src/main/java/org/radarcns/util/RecordConverterFactory.java

```diff
@@ -31,4 +49,70 @@
      * @throws IOException if the converter could not be created
      */
     RecordConverter converterFor(Writer writer, GenericRecord record, boolean writeHeader) throws IOException;
+
+    default boolean hasHeader() {
+        return false;
+    }
+
+    default void sortUnique(Path path) throws IOException {
+        SortedSet<String> sortedLines = new TreeSet<>();
```

Comment:
Hopefully, this method will typically process already sorted files or files with a relatively small number of anomalies. In that case using ArrayList.sort + extra check for duplicates will be much faster and less memory consuming than TreeSet. Probably, it would even make sense to always assume optimistically that the input file is already sorted and contains no duplicates, search for anomalies keeping only two subsequent lines in memory and only read the full file in memory and create a new file if any anomaly found.